# Data collection:

In [ ]:
!pip install google-search-results


Collected data using SerpApi. https://serpapi.com/

In [ ]:
from serpapi import GoogleSearch


Hotels = []
for i in range(0,101,20):
  params = {
  "engine": "google_maps",
  "start": i,
  "q": "Dubai Hotels",
  "ll": "@40.7455096,-74.0083012,14z",
  "api_key": "ad944e728e7267dbfed3623b663cd23686ac14f1a9a9ccf42b1ca1947c4270ed"
}
  search = GoogleSearch(params)
  results_Hotels = search.get_dict()
  Hotels.append(results_Hotels)

collected Hotels data from google map in Dubai.

In [ ]:
import json
from google.colab import files

# Save dict as JSON file
with open("hotel_reviews.json", "w") as f:
    json.dump(Hotels, f, indent=4)

# Download the file
files.download("hotel_reviews.json")


<IPython.core.display.Javascript object>

<IPython.core.display.Javascript object>

Using those hotel names, we will extract the placeId for search engine to scrap reviews.

In [ ]:
# len(Hotels[0]['local_results'])

lst_place_id = []

for i in range(len(Hotels)):
  for j in range(len(Hotels[i]['local_results'])):
    lst_place_id.append(Hotels[i]['local_results'][j]['place_id'])

print(lst_place_id)

['ChIJoYrk4C5oXz4Rz8zfVMtxSpU', 'ChIJowBFSu91Xz4RLS9kcNQWxE8', 'ChIJo22Ljw1rXz4RkciYKT0agT8', 'ChIJbTHIgIEVXz4RIPXU6nFB2Gs', 'ChIJyer95yhoXz4Rwl9l67jIHT4', 'ChIJVeSGVytDXz4R3tA72dWz_WE', 'ChIJAauOmZhdXz4RkJgo1j-UjyQ', 'ChIJyXyHcABqXz4R48N4-tR2CgM', 'ChIJp5gj9o1CXz4Rde77RZWQvKs', 'ChIJp8w6rTtDXz4RyfTyXTjXAU8', 'ChIJr7ycDMxcXz4R-ZfzIBvQjaY', 'ChIJo7DaJI1pXz4RScW1ssHBr1o', 'ChIJ4bjdQYNCXz4RUN7vxDANHsU', 'ChIJMZZ9A5lDXz4RsJmswWvSeFs', 'ChIJM7ZvJ4lCXz4RzogZiHR7Dn0', 'ChIJjcCRoiJDXz4R_bNfEdvAPVM', 'ChIJURh7NutCXz4Remv0bkTdQCw', 'ChIJAauOmZhdXz4RKs_nETzgBDc', 'ChIJvSwT-ZJCXz4RkKTQzvKWOmU', 'ChIJR6olL3BdXz4RhPxuIANrT-U', 'ChIJBWNFnphdXz4REguXwMzcqUQ', 'ChIJBWNFnphdXz4RzFTpwCSdnz4', 'ChIJMRLSra5pXz4RsYLSAgxjZvw', 'ChIJWW01m8hbXz4R8IpbVL6w9ZQ', 'ChIJj-fWdABdXz4RXQD5gORu3hI', 'ChIJUzfYgMhcXz4R__--nS9dH2s', 'ChIJ0QItrPJbXz4RfvBQdD-SXOU', 'ChIJxYvl8otCXz4Rb7WNVpc6p_0', 'ChIJEUdJHJ5dXz4RFE06gPYGxmg', 'ChIJm5AF9LxDXz4R-x__BNLedrM', 'ChIJ_SkQ_C9aXz4RxM5m6whBis8', 'ChIJh7WXy85cXz4RWN-OwLu4aF0', 'ChIJZz

In [ ]:
Hotels[0]['local_results'][0]['place_id']

'ChIJoYrk4C5oXz4Rz8zfVMtxSpU'

In [ ]:
for i in range(0,101,20):
  print(i)

0
20
40
60
80
100


In [ ]:
results_Hotels['local_results'][0]['place_id']

# lst_place_id = []

# for i in range(len(results_Hotels['local_results'])):
#   lst_place_id.append(results_Hotels['local_results'][i]['place_id'])

'ChIJBWNFnphdXz4REguXwMzcqUQ'

In [ ]:
len(lst_place_id)

112

We got 112 different hotels.

### collecting reviews and saving them:

It will use place_id as a search to get reviews from that particular hotel.

In [ ]:
from serpapi import GoogleSearch



Review_lst = []

for i in range(len(lst_place_id)):
  params = {
  "engine": "google_maps_reviews",
  "place_id": lst_place_id[i],
  "hl": "en",
  "api_key": "ad944e728e7267dbfed3623b663cd23686ac14f1a9a9ccf42b1ca1947c4270ed"
}
  search = GoogleSearch(params)
  results = search.get_dict()
  Review_lst.append(results)


In [ ]:
import json
from google.colab import files

# Save dict as JSON file
with open("hotel_reviews_maps.json", "w") as f:
    json.dump(Review_lst, f, indent=4)

# Download the file
files.download("hotel_reviews_maps.json")


<IPython.core.display.Javascript object>

<IPython.core.display.Javascript object>

In [ ]:
# len(Review_lst[0]['snippet'])

Final_G_Rev = []

Review_lst[0]['reviews'][0]['snippet']

for i in range(len(Review_lst)):

  if 'reviews' in Review_lst[i]:
    review = Review_lst[i]['reviews']

    for j in range(len(Review_lst[i]['reviews'])):
      if 'snippet' in review[j]:
        Final_G_Rev.append(Review_lst[i]['reviews'][j]['snippet'])

In [ ]:
len(Final_G_Rev)

860

In [ ]:
import json
from google.colab import files

# Save dict as JSON file
with open("Final_reviews.json", "w") as f:
    json.dump(Final_G_Rev, f, indent=4)

# Download the file
files.download("Final_reviews.json")

<IPython.core.display.Javascript object>

<IPython.core.display.Javascript object>